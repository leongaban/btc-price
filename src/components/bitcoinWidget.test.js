import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

import BitcoinWidget from './bitcoinWidget';

let wrapper;

const props = {
  logo: 'foo',
  coin: {
    price: 0
  },
  refresh: jest.fn()
};

describe('<BitcoinWidget /> layout', () => {

  beforeEach(() => {
    wrapper = shallow(<BitcoinWidget {...props}/>);
  });

  it('renders correct markup', () => {
    expect(wrapper.find('header').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
  });

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
