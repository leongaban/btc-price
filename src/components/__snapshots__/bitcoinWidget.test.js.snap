// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BitcoinWidget /> layout should render a component matching the snapshot 1`] = `
<div
  className="bitcoin-wrapper shadow"
>
  <header>
    <img
      alt="Bitcoin Logo"
      src="foo"
    />
  </header>
  <div
    className="price"
  >
    Coinbase $
    0
  </div>
  <button
    className="btn striped-shadow white"
    onClick={[Function]}
  >
    <span>
      Refresh
    </span>
  </button>
</div>
`;
